chain/network/src/stats/metrics.rs

```diff
@@ -6,25 +6,84 @@ use near_metrics::{
 };
 use near_network_primitives::types::{PeerType, RoutedMessageBody};
 use once_cell::sync::Lazy;
-use std::collections::HashMap;
 
-static PEER_CONNECTIONS: Lazy<IntGaugeVec> = Lazy::new(|| {
-    near_metrics::try_create_int_gauge_vec(
-        "near_peer_connections",
-        "Number of connected peers",
-        &["peer_type", "encoding"],
-    )
-    .unwrap()
-});
+/// Labels represents a schema of an IntGaugeVec metric.
+pub trait Labels: 'static {
+    /// Names of the gauge vector labels.
+    fn names() -> Vec<&'static str>;
+    /// Converts self to a list of label values.
+    /// values().len() should be always equal to names().len().
+    fn values(&self) -> Vec<&'static str>;
+}
 
-pub(crate) fn set_peer_connections(values: HashMap<(PeerType, Option<Encoding>), i64>) {
-    for ((pt, enc), v) in values {
-        PEER_CONNECTIONS
-            .with_label_values(&[pt.into(), enc.map(|e| e.into()).unwrap_or("unknown")])
-            .set(v);
+/// Type-safe wrapper of IntGaugeVec.
+pub struct Gauge<L: Labels> {
+    inner: IntGaugeVec,
+    _labels: std::marker::PhantomData<L>,
+}
+
+pub struct GaugePoint<L: Labels> {
+    gauge: &'static Gauge<L>,
+    point: IntGauge,
+}
+
+impl<L: Labels> Gauge<L> {
+    /// Constructs a new prometheus Gauge with schema `L`.
+    pub fn new(name: &str, help: &str) -> Self {
+        Self {
+            inner: near_metrics::try_create_int_gauge_vec(name, help, &L::names()).unwrap(),
```

Comment:
gut feeling is to push this `.unwrap` to the call-site.

Reply:
done